### PR TITLE
Closes #472 - Don't panic if not in a repository

### LINKED ISSUE #472: Shorter error message when opening lazygit in a non-repo folder
**Describe the bug**

**To Reproduce**
1. Go to any folder that is not a git repository (and neither are its parent folders).
2. `$ lazygit`
3. See error:
```
2019/07/04 12:45:01 An error occurred! Please create an issue at https://github.com/jesseduffield/lazygit/issues

*errors.errorString fatal: not a git repository (or any of the parent directories): .git

/private/tmp/lazygit-20190512-43735-1v1doie/lazygit-0.8/src/github.com/jesseduffield/lazygit/pkg/commands/os.go:150 (0x15e1d64)
/private/tmp/lazygit-20190512-43735-1v1doie/lazygit-0.8/src/github.com/jesseduffield/lazygit/pkg/commands/os.go:63 (0x15e142d)
/private/tmp/lazygit-20190512-43735-1v1doie/lazygit-0.8/src/github.com/jesseduffield/lazygit/pkg/commands/os.go:116 (0x1838ddd)
/private/tmp/lazygit-20190512-43735-1v1doie/lazygit-0.8/src/github.com/jesseduffield/lazygit/pkg/commands/os.go:116 (0x1838dbf)
/private/tmp/lazygit-20190512-43735-1v1doie/lazygit-0.8/src/github.com/jesseduffield/lazygit/pkg/app/app.go:119 (0x18389c0)
/private/tmp/lazygit-20190512-43735-1v1doie/lazygit-0.8/src/github.com/jesseduffield/lazygit/main.go:48 (0x18398fd)
/usr/local/Cellar/go/1.12.5/libexec/src/runtime/proc.go:200 (0x102f34c)
/usr/local/Cellar/go/1.12.5/libexec/src/runtime/asm_amd64.s:1337 (0x105a0f1)
```

**Expected behavior**
A shorter error message, for example:
```
$ git status
fatal: not a git repository (or any of the parent directories): .git
```

**Desktop (please complete the following information):**
 - OS: macOS 10.13.6 (17G7024)
 - Lazygit Version 0.8

### PR DESCRIPTION
**Pull request by giorgioprevitera**
Display a friendly message and exit with an error if not
in a Git repository. Using the same approach used in this PR:
https://github.com/jesseduffield/lazydocker/pull/14/files

### CODE REVIEW

File: pkg/i18n/english.go [hunk 1]
```diff
@@ -771,6 +771,9 @@ func addEnglish(i18nObject *i18n.Bundle) error {
 		}, &i18n.Message{
 			ID:    "stashOptions",
 			Other: "Stash options",
+		}, &i18n.Message{
+			ID:    "notARepository",
+			Other: "Error: must be run inside a repository",
```
Comment (mjarkk) [inline pkg/i18n/english.go]: I think "Error: must be run inside a **git** repository" is a better choice, there are more things that are called a repository.

File: pkg/app/app.go [hunk 2]
```diff
@@ -158,7 +163,8 @@ func (app *App) Run() error {
 		os.Exit(0)
 	}
 
-	return app.Gui.RunWithSubprocesses()
+	err := app.Gui.RunWithSubprocesses()
```
Comment (mjarkk) [inline pkg/app/app.go]: Why not just `return app.Gui.RunWithSubprocesses()`?

File: pkg/i18n/english.go (hunk 1 above)
Comment (giorgioprevitera) [reply pkg/i18n/english.go]: True, updating

File: pkg/app/app.go (hunk 2 above)
Comment (giorgioprevitera) [reply pkg/app/app.go]: I agree, I was just following this approach, happy to change it back: https://github.com/jesseduffield/lazydocker/pull/14/files#diff-637a60a871f8b69622bdfd3d3b2fcad1

Comment (mjarkk) [reply pkg/app/app.go]: I think that is there for later use or jesseduffield forgot to change it back

Comment (giorgioprevitera) [reply pkg/app/app.go]: Do you think we should leave it as it is so the two projects are in sync, given that they are pretty similar?

Comment (giorgioprevitera) [reply pkg/app/app.go]: Hi @mjarkk are you happy to merge this?

**Comment by jesseduffield** [comment]
tested and it works! I think @mjarkk is on holiday so I'll assume he's fine with this PR as-is. Merging :)

**Comment by giorgioprevitera** [comment]
this PR closes #472

**Pull request merged**